pixel-saver@deadalnix.me/decoration.js

```diff
@@ -71,7 +71,44 @@ function guessWindowXID(win) {
 			}
 		}
 	}
-	
+
+	// Try enumerating all available windows and match the title. Note that this
+	// may be necessary if the title contains special characters and `x-window`
+	// is not available.
+	let result = GLib.spawn_command_line_sync('xprop -root _NET_CLIENT_LIST');
+	LOG('xprop -root _NET_CLIENT_LIST')
+	if (result[0]) {
+		let str = result[1].toString();
+
+		// Get the list of window IDs.
+		let windowList = str.match(/0x[0-9a-f]+/g);
+
+		// For each window ID, check if the title matches the desired title.
+		for (var i = 0; i < windowList.length; ++i) {
+			let cmd = 'xprop -id "' + windowList[i] + '" _NET_WM_NAME _PIXEL_SAVER_ORIGINAL_STATE';
+			let result = GLib.spawn_command_line_sync(cmd);
+			LOG(cmd)
+
+			if (result[0]) {
+				let output = result[1].toString();
+				let isManaged = output.indexOf("_PIXEL_SAVER_ORIGINAL_STATE(CARDINAL)") > -1;
+
+				if (isManaged) {
+					continue;
+				}
+
+				let title = output.substring(output.indexOf('"') + 1, output.length - 2);
```

Comment:
The one `xprop` command matches `_NET_WM_NAME` as well as `_PIXEL_SAVER_ORIGINAL_STATE`, so you only need to run a command once for both pieces of information.